arcs_fact_tests/fail_different_tag.dl

```diff
@@ -0,0 +1,31 @@
+// Original Arcs manifest:
+// Fails when different tag is claimed.
+// #Ingress: P1
+// #Fail: hc:P2.bar is trusted
+//particle P1
+//  foo: writes Foo {}
+//  claim foo is notTrusted
+//particle P2
+//  bar: reads Foo {}
+//  check bar is trusted
+//recipe R
+//  P1
+//    foo: writes h
+//  P2
+//    bar: reads h
+#include "taint.dl"
+#include "analyze_tag_checks_test_helper.dl"
+
+claimHasTag("R.P1.foo", "notTrusted").
+edge("R.P1.foo", "R.h.Foo").
+edge("R.h.Foo", "R.P2.bar").
+
+// This check was in the original manifest. We expect it not to pass.
+checkTag("R.P2.bar", "trusted", "PRESENT").
+
+// This check was not in the original manifest, but we add it to make sure the dataflow did
+// something.
+checkTagExtra("R.P2.bar", "notTrusted", "PRESENT").
+
```

Comment:
I am finding it a bit hard to follow all the different expectation checks and glean what is being tested. Here is a possible approach. Let us define a relation called `testFails(message: symbol)` and use it to capture the various expectations from the test. The tests would look like the following now:

```
testFails("`check R.P2.bar is trusted` should not be true") :- 
  mayHaveTag("R.P2.bar", "trusted")
testFails("`notTrusted` tag is not present for `R.P2.bar`):-
  !mayHaveTag("R.P2,bar", "notTrusted")
```

Reply:
I see, ok, that makes sense.

Reply:
Done.